internal/cli/atlas/quickstart/quick_start.go

```diff
@@ -0,0 +1,357 @@
+	return &atlas.ProviderSettings{
+		InstanceSizeName:    tier,
+		ProviderName:        tenant,
+		RegionName:          opts.Region,
+		BackingProviderName: opts.Provider,
+	}
+}
+
+func (opts *Opts) askClusterOptions() error {
+	var qs []*survey.Question
+
+	message := "Insert the cluster name"
+	clusterName := opts.ClusterName
+	if clusterName == "" {
+		clusterName = opts.newClusterName()
+		if clusterName != "" {
+			message = fmt.Sprintf("Insert the cluster name [Press Enter to use the auto-generated name '%s']", clusterName)
+		}
+		qs = append(qs, newClusterNameQuestion(clusterName, message))
+	}
+
+	if opts.Provider == "" {
+		qs = append(qs, newClusterProviderQuestion())
+	}
+
+	if err := survey.Ask(qs, opts); err != nil {
+		return err
+	}
+
+	if regionQ := newRegionQuestions(opts.Region, opts.Provider); regionQ != nil {
+		// we call survey.Ask two times because the region question needs opts.Provider to be populated
+		return survey.Ask([]*survey.Question{regionQ}, opts)
+	}
+
+	return nil
+}
+
+// askDBUserAccessListOptions allows the user to set required flags by using interactive prompts
+func (opts *Opts) askDBUserAccessListOptions() error {
+	var qs []*survey.Question
+
+	message := "Insert the Username for authenticating to MongoDB"
+	dbUser := opts.DBUsername
+	if dbUser == "" {
+		dbUser = dbUsername()
+		if dbUser != "" {
+			message = fmt.Sprintf("Insert the Username for authenticating to MongoDB [Press Enter to use '%s']", dbUser)
+		}
+
+		qs = append(qs, newDBUsernameQuestion(dbUser, message, opts.validateUniqueUsername))
+	}
+
+	if opts.DBUserPassword == "" {
+		qs = append(qs, newDBUserPasswordQuestion())
+	}
+
+	if len(opts.IPAddresses) == 0 {
+		message = "Insert the IP entry to add to the Access List"
+		publicIP := net.IPAddress()
+		if publicIP != "" {
+			message = fmt.Sprintf("Insert the IP entry to add to the Access List [Press Enter to use your public IP address '%s']", publicIP)
+		}
+		q := newAccessListQuestion(publicIP, message)
+		qs = append(qs, q)
+	}
+
+	if len(qs) > 0 {
+		if err := survey.Ask(qs, opts); err != nil {
+			return err
+		}
+	}
+
+	if len(opts.IPAddresses) == 0 && opts.IPAddress != "" {
+		opts.IPAddresses = []string{opts.IPAddress}
+	}
+
+	if opts.DBUserPassword == "" {
+		// The user wants to auto-generate the password
+		pwd, err := randgen.GenerateRandomBase64String(passwordLength)
+		if err != nil {
+			return err
+		}
+		opts.DBUserPassword = pwd
+	}
+
+	return nil
+}
+
+func (opts *Opts) validateUniqueUsername(val interface{}) error {
+	username, ok := val.(string)
+	if !ok {
+		return errors.New("the username is not valid")
+	}
+
+	_, err := opts.store.DatabaseUser(convert.AdminDB, opts.ConfigProjectID(), username)
+	var target *atlas.ErrorResponse
+
+	if err != nil && errors.As(err, &target) {
+		if target.ErrorCode == "USERNAME_NOT_FOUND" {
+			return nil
+		}
+		return err
+	}
+
+	return errors.New("a user with this username already exists")
```

Comment:
[nit] please let add the username to the error, it may be helpful if the person doesn't remember what they typed or even if they had a typo